tinygrad/codegen/linearizer.py

```diff
@@ -402,7 +403,8 @@ def ssa(name, ltype=dtypes.float) -> Token:
           self.uop(UOps.BARRIER, None, [], ())
 
         # load earlybufs
-        loaded_buffers.update({b:self.global_load(self.bufs.index(self.local_alias[i]) if i in self.local_alias else i, global_idxs+local_idxs+reduce_idxs+full_upcast_idxs) for i,b in enumerate(self.bufs) if b in self.earlybufs and i != 0})
+        earlybuf_cache: Dict[Any, Dict[str, Token]] = defaultdict(dict)
```

Comment:
Do you need separate caches?

Reply:
I ran into a case somewhere in the tests where a there was the same load in earlybufs and latebufs, but the earlybufs load was in the loop and out of scope of the late ast.

Reply:
kk, two cache is fine